asteroid/utils/torch_utils.py

```diff
@@ -123,3 +123,107 @@ def are_models_equal(model1, model2):
         if p1.data.ne(p2.data).sum() > 0:
             return False
     return True
+
+
+def slice(sources, w_size=16384, h_size=16384 // 2):
+    """  Slice a signal in slices of size w_size
+
+     Args:
+         sources (torch.Tensor): sources of shape [nb_sources,channels,samples]
+         w_size (int): size of the window
+         h_size (int): size of the stride
+     Returns:
+         sliced signal (torch.Tensor): shape [nb_sources,nb_slices,channels,
```

Comment:
At inference, we are expecting signals with shape (1, 1, wal_len), (1, wav_len) or (wav_len,) but we might not want to make the assumption the shape is always (*, *, time).

Reply:
A recursive solution would work. Or a class interface that handles the reshaping.

Reply:
If we want to support several input shapes, their inversion should be unique (unslice(slice(x)) should equal x).

Reply:
I think i'm going to try the class interface.